src/lib/components/workflow/workflow-relationships.svelte

```diff
@@ -8,23 +8,23 @@
   import Badge from '$lib/holocene/badge.svelte';
   import ChildWorkflowsTable from '$lib/components/workflow/child-workflows-table.svelte';
   import WorkflowDetail from '$lib/components/workflow/workflow-detail.svelte';
+  import type { CombinedAttributes } from '$lib/utilities/format-event-attributes';
+
+  type WorkflowEventWithAttributes = WorkflowEvent & {
```

Comment:
Would the better solution be to update `attributes` on `WorkflowEvent` in `events.d.ts`? I'm not sure `EventAttribute` is always accurate 🤔

Reply:
The workflow event types are tricky, I'm not sure I have the expertise to recommend the best approach, but I do think we should avoid casting those `lastEvent` and `firstEvent` reactive variables if possible. We may be able to use the `EventWithAttributes` type and/or use some of the [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) in `is-event-type.ts`. @stevekinney would prob have the best advice here 🙂

Reply:
I'm thinking something like:

```svelte
let firstEvent: unknown
$: firstEvent = $eventHistory.start[0];
if (isWorkflowExecutionStartedEvent(firstEvent)) {
  // do the thing with firstEvent types as `WorkflowExecutionStartedEvent`
}
```

Idk if that's actually helpful but just some thoughts 😅

Reply:
Yea, that casting is opting out of a lot of type safety. Also: Do you want the _first_ event or do you want the `WorkflowExecutionStarted` event? (Yes, I get that the first event probably is the `WorkflowExecutionStarted`, but that's more of a coincidence than your intention in the code.

Reply:
This is alway _way_ too much logic to live in the view layer.